PYME/Analysis/points/ripleys.py

```diff
@@ -158,10 +239,27 @@ def ripleys_k(x, y, n_bins, bin_size, mask=None, bbox=None, z=None, threaded=Fal
     yu = yu.ravel()
     if zu is not None:
         zu = zu.ravel()
-            
-    return ripleys_k_from_mask_points(x=x, y=y, z=z,
+
+    bb, K = ripleys_k_from_mask_points(x=x, y=y, z=z,
                                       xu=xu, yu=yu, zu=zu,
                                       n_bins=n_bins, bin_size=bin_size, mask_area=mask_area, threaded=threaded)
+
+    if mc:
+        # Monte-Carlo simulations on the mask
+        K_arr = np.zeros((n_sim,len(K)))
+        for _i in range(n_sim):
+            xm, ym, zm, _ = mc_points_from_mask(mask, len(x), three_d, coord_origin)
+            _, K_arr[_i,:] = ripleys_k_from_mask_points(x=xm, y=ym, z=zm,
+                                      xu=xu, yu=yu, zu=zu,
+                                      n_bins=n_bins, bin_size=bin_size, mask_area=mask_area, threaded=threaded)
+        # Envelope
+        K_min = np.min(K_arr, axis=0)
+        K_max = np.max(K_arr, axis=0)
```

Comment:
An envelope would more normally be +- 1 sigma or (probably better) +- one 95% CI, not the min and max of the simulation.

Reply:
Will change this.